### PR TITLE
implements PATCH /api/articles/:article_id

### PR DESCRIPTION
**Pull request by steelfish1010**

### CODE REVIEW

**Comment by ghost** [comment]
Nice work Ali, your tests look great! You may wish to consider what if your `send(articleUpdate)` was sent with empty `inc_votes`.

File: models/articles.model.js [hunk 1]
```diff
@@ -13,3 +13,20 @@ exports.fetchArticleById = async (article_id) => {
...
+exports.updateArticle = async (body, article_id) => {
+  const votes = parseInt(body.inc_votes);
+  // console.log(votes, "<-- votes in model");
+  const res = await db.query(
+    `
+  UPDATE articles
+  SET 
+  votes = votes + $1
+  WHERE article_id = $2
+  RETURNING *
```
Comment (dentednerd) [inline models/articles.model.js]: Don't forget your semicolon at the end of an SQL query string

File: routers/api-router.js [hunk 2]
```diff
@@ -9,8 +12,13 @@ apiRouter
   })
   .get("/topics", getTopics)
   .get("/articles/:article_id", getArticleById)
+  .patch("/articles/:article_id", patchArticle)
   .all("/*", (req, res) => {
     res.status(404).send({ msg: "Path not found" });
   });
 
+apiRouter.use((err, req, res, next) => {
+  console.log(err, "<-- err in apiRouter");
```
Comment (dentednerd) [inline routers/api-router.js]: Always a good idea to remove console.logs as soon as you don't need them anymore to keep your code nice and clean

File: models/articles.model.js (hunk 1 above)
Comment (steelfish1010) [reply models/articles.model.js]: Done :)

File: routers/api-router.js (hunk 2 above)
Comment (steelfish1010) [reply routers/api-router.js]: Done :)

**Comment by steelfish1010** [comment]
> 

Hi, not sure what you mean by this one. If I try to test with { inc_votes: } I get a syntax error and the test won't run. Can you expand pls?

File: routers/api-router.js [hunk 3]
```diff
@@ -9,8 +12,13 @@ apiRouter
   })
   .get("/topics", getTopics)
   .get("/articles/:article_id", getArticleById)
+  .patch("/articles/:article_id", patchArticle)
   .all("/*", (req, res) => {
     res.status(404).send({ msg: "Path not found" });
   });
 
+apiRouter.use((err, req, res, next) => {
+  // console.log(err, "<-- err in apiRouter");
```
Comment (dentednerd) [inline routers/api-router.js]: Remove this console.log please!

File: models/articles.model.js [hunk 4]
```diff
@@ -13,3 +13,20 @@ exports.fetchArticleById = async (article_id) => {
...
+    `
+  UPDATE articles
+  SET 
+  votes = votes + $1
+  WHERE article_id = $2
+  RETURNING *;
+  `,
+    [votes, article_id]
+  );
+  // console.log(res.rows[0], "<-- res.rows[0] in model");
```
Comment (dentednerd) [inline models/articles.model.js]: Remove this console.log please!

File: models/articles.model.js [hunk 5]
```diff
@@ -13,3 +13,20 @@ exports.fetchArticleById = async (article_id) => {
     return res.rows[0];
   }
 };
+
+exports.updateArticle = async (body, article_id) => {
+  const votes = parseInt(body.inc_votes);
+  // console.log(votes, "<-- votes in model");
```
Comment (dentednerd) [inline models/articles.model.js]: Remove this console.log please!

File: controllers/articles.controller.js [hunk 6]
```diff
@@ -9,3 +9,20 @@ exports.getArticleById = async (req, res, next) => {
...
+    const { body } = req;
+    const dbQueries = [
+      updateArticle(body, article_id),
+      fetchArticleById(article_id),
+    ];
+    const results = await Promise.all(dbQueries);
+    const updatedArticle = results[0];
+    res.status(200).send({ updatedArticle });
+  } catch (err) {
+    // console.log(err, "<-- err in controller");
```
Comment (dentednerd) [inline controllers/articles.controller.js]: Remove this console.log please!

File: routers/api-router.js (hunk 3 above)
Comment (steelfish1010) [reply routers/api-router.js]: done

File: models/articles.model.js (hunk 4 above)
Comment (steelfish1010) [reply models/articles.model.js]: done

File: models/articles.model.js (hunk 5 above)
Comment (steelfish1010) [reply models/articles.model.js]: done

File: controllers/articles.controller.js (hunk 6 above)
Comment (steelfish1010) [reply controllers/articles.controller.js]: done

**Review by HeatherEver — COMMENTED** [review]
Looks good! ⭐

**Pull request merged**